Applications/Spire/Include/Spire/KeyBindings/KeyBindings.hpp

```diff
@@ -227,11 +290,39 @@ namespace Details {
       std::vector<KeyBindingMapping> m_bindings;
   };
 
+  template<typename T>
+  KeyBindings::Tag::Tag(int key, T value)
+    : Tag(key, boost::optional<T>(std::move(value))) {}
+
+  template<typename T>
+  KeyBindings::Tag::Tag(int key, boost::optional<T> value)
+    : m_key(key),
+      m_value(std::move(value)) {}
+
+  template<typename T>
+  boost::optional<T> KeyBindings::Tag::get_value() const {
+    return std::visit(
+      [] (auto& value) {
+        if constexpr(std::is_same_v<std::decay_t<decltype(value)>,
+            boost::optional<T>>) {
+          return value;
+        } else {
+          return boost::optional<T>();
```

Comment:
Don't think a silent failure works well, I'd rather an exception/abort. What do you think?

Reply:
Yes, returning a nullopt is somewhat random and can really conceal issues. Especially for implicitly convertible types, e.g. in:

```
auto tag = Tag(1, 10);
auto value = tag.get_value<double>();
```

There isn't a lot of exception handling code in Spire, so I am not sure about the convention in this regard. As an option, I can throw `std::bad_cast` here.